themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,36 +28,35 @@
   </div>
 <?php endif; ?>
-<div class="lightbox-scroll full-facets">
+
+<!-- Hopefully we can find a better way for passthrough of these variables -->
+<?php if (method_exists($this->results->getParams(), 'setFacetContains')): ?>
+  <input class="ajax_param" data-name="contains" placeholder="<?=$this->transEscAttr('Filter')?>" class="form-control" type="text" value="<?=htmlspecialchars($this->contains)?>">
+<?php endif; ?>
+<input class="ajax_param" data-name="facet" type="hidden" value="<?=$this->facet?>">
+<input class="ajax_param" data-name="facetexclude" type="hidden" value="<?=$this->exclude?>">
+<input class="ajax_param" data-name="facetop" type="hidden" value="<?=$this->operator?>">
+<input class="ajax_param" data-name="facetpage" type="hidden" value="<?=$this->page?>">
+<input class="ajax_param" data-name="facetsort" type="hidden" value="<?=$this->sort?>">
+<input class="ajax_param" data-name="urlBase" type="hidden" value="<?=$urlBase?>">
+<input class="ajax_param" data-name="searchAction" type="hidden" value="<?=$searchAction?>">
+<input class="ajax_param" data-name="ajaxLightbox" type="hidden" value="<?=$this->ajaxLightbox ? intval($this->ajaxLightbox) : intval($this->inLightbox) ?>">
+
+
+<div id="facet-info-result" class="lightbox-scroll full-facets">
   <?php foreach ($this->sortOptions as $key => $sort): ?>
     <?php $active = $this->sort == $key; ?>
-    <div class="full-facet-list facet-group<?php if (!$active): ?> hidden<?php endif; ?>" id="facet-list-<?=$this->escapeHtmlAttr($key) ?>">
-      <?php if ($active): ?>
-        <?php if ($this->page > 1): ?>
-          <a href="<?=$urlBase . '&amp;facetpage=' . ($this->page - 1) . '&amp;facetsort=' . urlencode($this->sort) ?>" class="facet js-facet-prev-page" data-page="<?=($this->page + 1) ?>" data-sort="<?=$this->sort ?>" data-limit="<?=count($this->data) ?>" data-lightbox-ignore><?=$this->translate('prev_ellipsis') ?></a>
-        <?php endif; ?>
-        <?php foreach ($this->data as $item): ?>
-          <?=$this->render('Recommend/SideFacets/single-facet.phtml', [
-            'exclude' => $this->exclude,
-            'facet' => $item,
-            'group' => $this->facet,
-            'url' => $this->results->getUrlQuery(),
-            'urlBase' => $searchAction,
-          ]) ?>
-        <?php endforeach; ?>
-      <?php endif; ?>
-      <?php if ($this->anotherPage): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=' . ($this->page + 1) . '&amp;facetsort=' . urlencode($key) ?>" class="facet js-facet-next-page" data-page="<?=($this->page + 1) ?>" data-sort="<?=$this->escapeHtmlAttr($key) ?>" data-lightbox-ignore>
-          <span class="text"><?=$this->translate('more_ellipsis') ?></span>
-        </a>
-      <?php endif; ?>
+    <div class="full-facet-list facet-group<?php if(!$active): ?> hidden<?php endif; ?>" id="facet-list-<?=$this->escapeHtmlAttr($key) ?>">
+      <!-- The content of this list will be generated dynamically, see facet-list-content.phtml -->
```

Comment:
Does this mean that the facet list ONLY renders by AJAX now, as opposed to rendering inline to start with and updating dynamically as needed? Probably not a huge problem if that's the case, but it will add extra HTTP requests that we might not need in the majority of situations.

Reply:
Good point! I just made a little change so that the initial load also renders the content directly instead of using another AJAX call for this.